src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTrigger.java

```diff
@@ -1849,23 +1852,27 @@ public synchronized void scheduled(ChangeBasedEvent event, ParametersAction para
         }
 
         /**
-         * Tries to cancel any jobs with the specified parameters. We look in
-         * both the build queue and currently executing jobs. This extra work is
-         * required due to race conditions when calling Future.cancel() - see
+         * Tries to cancel any job, which was triggered by the given change event.
+         * <p>
+         * Since the event is always noted in the build cause, it is easy to
+         * identify which specific builds shall be cancelled, without having
+         * to dig down into the parameters, which might've been mutated by the
+         * build while it was running. (This was the previous implementation)
+         * <p>
+         * We look in both the build queue and currently executing jobs.
+         * This extra work is required due to race conditions when calling
+         * Future.cancel() - see
          * https://issues.jenkins-ci.org/browse/JENKINS-13829
          *
-         * @param parameters
-         *            The parameters to match against.
+         * @param event
+         *            The event that originally triggered the build.
          */
-        private void cancelJob(ParametersAction parameters) {
+        private void cancelJob(GerritTriggeredEvent event) {
             // Remove any jobs in the build queue.
-            List<hudson.model.Queue.Item> itemsInQueue = Queue.getInstance().getItems(job);
-            for (hudson.model.Queue.Item item  : itemsInQueue) {
-                List<ParametersAction> params = item.getActions(ParametersAction.class);
-                for (ParametersAction param : params) {
-                    if (param.equals(parameters)) {
-                        Queue.getInstance().cancel(item);
-                    }
+            List<hudson.model.Queue.Item> itemsInQueue = Queue.getInstance().getItems(myProject);
```

Comment:
Thanks, will fix it ASAP. That's what you get for re-basing and just looking at the mergetool. :P